Assets/ChessStates.cs

```diff
@@ -14,14 +14,18 @@ boolean validateMove(int startIndex, int endIndex)
             First move: +- 8 or +- 16
 
         */
-        if(endIndex >= startIndex + 7 && endIndex <= startIndex + 9)
-        return false;
```

Comment:
Ok so for the pawn movement, I think we have the logic a little messed up here. Wasn't intending for this section to have a bug but when considering the chess board, this is how the piece movement works:

We have an 8x8 board where each matrix cell has a number from 0 -> 63.

The startIndex is the index the piece begins on, and the endIndex is where the piece is placed.

We know that pawns can move diagonally 1 square in the left and right direction, as well as up 1 square in the forward direction.

If we were to imagine for instance that our pawn starts on square 3, his valid moves would be the square immediately on top of him, and the square to the right and left diagonally. Mathematically this works out to startIndex +7 (top left) startIndex + 8 (top) startIndex +9 (top right).

If our pawn's endIndex falls in that range, we want to return true, not false. For now, let's just validate the moves for whichever piece color starts on the bottom. We can work on distinguishing the piece colors later on.